### PR TITLE
Documentation for receiving of custom properties from IoT Hub through endpoint that is compatible with Event Hubs

### PR DESCRIPTION
**Pull request by me11dray**
Documentation for [Receiving of custom properties from IoT Hub through endpoint that is compatible with Event Hubs](https://github.com/dapr/components-contrib/pull/2298)

Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [x] [Read the contribution guide](https://docs.dapr.io/contributing/docs-contrib/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

This includes documentation for receiving Application Properties (enriched messages) from an Event hub . The feature ensured enriched messages containing additional properties are forwarded as headers.

Feature PR : [Receiving of custom properties from IoT Hub through endpoint that is compatible with Event Hubs](https://github.com/dapr/components-contrib/pull/2298)

Also can act as a reference for [#6075 : Propagate arbitrary/configurable http headers from publisher to subscribers](https://github.com/dapr/dapr/issues/6075)

### CODE REVIEW

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-eventhubs.md [hunk 1]
```diff
@@ -180,6 +180,47 @@ For example, the headers of a delivered HTTP subscription message would contain:
...
+  routes: 
+    default: /checkout
+  pubsubname: order-pub-sub
+  metadata:
+     requireAllProperties: "true"
+scopes:
+- orderprocessing
+- checkout
+```
+This ensures enriched messages containing additional properties are forwarded as headers. For example, the device-to-cloud events created by Azure IoT Hub devices can contain user defined additional [IoT Hub Application Properties](https://learn.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-messages-construct#application-properties-of-d2c-iot-hub-messages), and the Azure Event Hubs pubsub component for Dapr will return the following as part of the response metadata ( in this example `iothub-creation-time-utc` is application property ):
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-eventhubs.md]:
```suggestion
This ensures enriched messages containing additional properties are forwarded as headers. For example, the device-to-cloud events created by Azure IoT Hub devices can contain user defined additional [IoT Hub Application Properties](https://learn.microsoft.com/azure/iot-hub/iot-hub-devguide-messages-construct#application-properties-of-d2c-iot-hub-messages). The Azure Event Hubs pub/sub component for Dapr returns the following as part of the response metadata (in this example, `iothub-creation-time-utc` is application property):
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-eventhubs.md [hunk 2]
```diff
@@ -180,6 +180,47 @@ For example, the headers of a delivered HTTP subscription message would contain:
   'traceparent': '00-4655608164bc48b985b42d39865f3834-ed6cf3697c86e7bd-01'
 }
 ```
+## Subscribing to Application Properties (enriched messages) from an Event hub
+
+It possible to get application properties (enriched messages) from an Event Hub by declaratively adding the `requireAllProperties` metadata entry to subscription specification.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-eventhubs.md]:
```suggestion
It is possible to get application properties (enriched messages) from an Event Hub by declaratively adding the `requireAllProperties` metadata entry to the subscription specification.
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-eventhubs.md [hunk 3]
```diff
@@ -180,6 +180,47 @@ For example, the headers of a delivered HTTP subscription message would contain:
...
+  routes: 
+    default: /checkout
+  pubsubname: order-pub-sub
+  metadata:
+     requireAllProperties: "true"
+scopes:
+- orderprocessing
+- checkout
+```
+This ensures enriched messages containing additional properties are forwarded as headers. For example, the device-to-cloud events created by Azure IoT Hub devices can contain user defined additional [IoT Hub Application Properties](https://learn.microsoft.com/azure/iot-hub/iot-hub-devguide-messages-construct#application-properties-of-d2c-iot-hub-messages). The Azure Event Hubs pub/sub component for Dapr returns the following as part of the response metadata (in this example, `iothub-creation-time-utc` is application property):
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-eventhubs.md]:
```suggestion
Adding this metadata entry ensures enriched messages containing additional properties are forwarded as headers. For example, the device-to-cloud events created by Azure IoT Hub devices can contain extra user-defined [IoT Hub Application Properties](https://learn.microsoft.com/azure/iot-hub/iot-hub-devguide-messages-construct#application-properties-of-d2c-iot-hub-messages). The Azure Event Hubs pub/sub component for Dapr returns the following as part of the response metadata (in this example, `iothub-creation-time-utc` is the application property):
```

**Review by hhunter-ms — CHANGES_REQUESTED** [review]

**Review by hhunter-ms — APPROVED** [review]

**Comment by msfussell** [comment]
@mecoding1 - You need to add an entry for `requireAllProperties` into the spec metadata table field https://docs.dapr.io/reference/components-reference/supported-pubsub/setup-azure-eventhubs/#spec-metadata-fields  providing a description and a link to the section you have added further down the document.

**Comment by me11dray** [comment]
> @mecoding1 - You need to add an entry for `requireAllProperties` into the spec metadata table field https://docs.dapr.io/reference/components-reference/supported-pubsub/setup-azure-eventhubs/#spec-metadata-fields providing a description and a link to the section you have added further down the document.

@msfussell Was wondering since we need to add `"requireAllProperties": "true" ` to the subscription metadata , while the section [https://docs.dapr.io/reference/components-reference/supported-pubsub/setup-azure-eventhubs/#spec-metadata-fields](https://docs.dapr.io/reference/components-reference/supported-pubsub/setup-azure-eventhubs/#spec-metadata-fields) refers to the component spec metadata fields ?

**Comment by msfussell** [comment]
> > @mecoding1 - You need to add an entry for `requireAllProperties` into the spec metadata table field https://docs.dapr.io/reference/components-reference/supported-pubsub/setup-azure-eventhubs/#spec-metadata-fields providing a description and a link to the section you have added further down the document.
> 
> @msfussell Was wondering since we need to add `"requireAllProperties": "true" ` to the subscription metadata , while the section https://docs.dapr.io/reference/components-reference/supported-pubsub/setup-azure-eventhubs/#spec-metadata-fields refers to the component spec metadata fields ?

@mecoding1 Missed that. Yes, does not need to be added to this table. But this then raise the question that subscription spec format applies to any topic subscription, so what happens when I apply this field to any pub/sub provider, not just Event Hubs? For example, say this field was on RabbitMQ pub/sub subscription, what would happen? This *may* need to be described here https://docs.dapr.io/developing-applications/building-blocks/pubsub/subscription-methods/ is this is the case.

**Comment by msfussell** [comment]
@mecoding1 - See my comment above on having this new field in the subscription topic. How does this apply to all components? Can you explain?

**Comment by me11dray** [comment]
> How does this apply to all components?

@msfussell  It is a one off field that is being utilized in the Azure Event Hub components i.e. for that particular subscription it will provide all the properties ( application + system ) . This is non-breaking change which doesn't interfere with the functionalities of the other components . But if such a case arises where we need additional metadata , we can always reutilize this field for the same purpose.

But atm, its only useful for Azure Event Hubs ( and IoT Hub ) component , but for other components it has no effect.

**Comment by me11dray** [comment]
@msfussell Do you think its ok to merge the PR or the details need to be added to the common subscription spec format other than in the current page ?

( since this applies to only the Event Hub subscription and not the others ? )

**Pull request closed without merging**